esiclient/tests/functional/utils.py

```diff
@@ -0,0 +1,68 @@
+#    Copyright (c) 2016 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import json
+
+from tempest.lib.common.utils import data_utils
+from tempest.lib import exceptions
+
+
+def node_create(client, driver='fake-hardware', name=None, params='',
+                fail_ok=False, merge_stderr=False):
+
+    if not name:
```

Comment:
We should probably be using data_utils.rand_name() regardless of whether or not one is specified to avoid conflicts with existing nodes on hardware these tests are to be run on. If a name is not specified, the name that is used in the CLI call will be something that looks like "tempest-baremetal-f45ab309cbb1". However, if a name is specified (e.g. "testing") the name used in the CLI call will just be "testing", rather than say, "tempest-testing-10a3bce11489f3".

Reply:
This still needs to be updated to prevent node name collisions, could probably look something like this:
```suggestion
    if not name:
        name = 'baremetal-node'
    name = data_utils.rand_name(name)
```

Reply:
This issue only happens if the user is passing in the name, right? If so, I don't think we should be randomizing that name; what if the test is specifically trying to test the creation of a node with the same name?